File: Verify This, Your Majesty/Queens.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Queens {

    public static void main(String args[]) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] coordinates = new int[n * 2];

        for (int i = 0; i < n * 2; i++) {
            String[] ip = br.readLine().split(" ");
            coordinates[i] = Integer.parseInt(ip[0]);
            coordinates[++i] = Integer.parseInt(ip[1]);

            for (int j = i - 2; j >= 0; j--) {
                int a = Math.abs(coordinates[i] - coordinates[j]);
                int b = Math.abs(coordinates[i - 1] - coordinates[--j]);

                if (a == b || a == 0 || b == 0) {
                    System.out.print("INCORRECT");
                    return;
                }
            }
        }
        
        System.out.print("CORRECT");
    }
}

File: Verify This, Your Majesty/Queens.cpp
// Problem: https://open.kattis.com/problems/queens

#include <iostream>
using namespace std;

int main() {
    
    int n;
    cin >> n;
    int coordinates[n * 2];
    
    for (int i = 0; i < n * 2; i++) {
        int x, y;
        cin >> x >> y;
        coordinates[i] = x;
        coordinates[++i] = y;
        
        for (int j = i - 2; j >= 0; j--) {
            int a = abs(coordinates[i] - coordinates[j]);
            int b = abs(coordinates[i - 1] - coordinates[--j]);
            
            if (a == b || a == 0 || b == 0) {
                cout << "INCORRECT";
                return 0;
            }
        }
    }
    
    cout << "CORRECT";
    return 0;
}
